casper/src/main/scala/coop/rchain/casper/Validate.scala

```diff
@@ -39,7 +39,10 @@ object Validate {
       block: BlockMessage,
       shardId: String,
       expirationThreshold: Int
-  ): F[ValidBlockProcessing] =
+  ): F[ValidBlockProcessing] = {
+    def validate(f: => Boolean, errorStatus: InvalidBlock): EitherT[F, InvalidBlock, ValidBlock] =
```

Comment:
You probably meant to make `errorStatus` as a thunk because boolean argument must be evaluated always.

Reply:
Not sure, what do you mean :thinking:

Reply:
I mean something like this where predicate `p` is not a thunk because it will be evaluated always when the function is called and parameter `errorStatus` will only be evaluated if `p` is `false` so it make sense to be defined as a thunk.
```scala
def validate(p: Boolean, errorStatus: => InvalidBlock)
```

For example here print will not be called.
```scala
def myPrintEager(p: Boolean, action: Unit) = if (p) action

def myPrintLazy(p: Boolean, action: => Unit) = if (p) action

myPrintEager(false, println("Lazily printing...")) // will print
myPrintLazy(false, println("Lazily printing...")) // will not print
```